tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -974,9 +976,12 @@ Status Converter::RenameAndMarkOutputTensors(
       MarkQuantizationRangesAsInferrable(tensor, layer->getOutput(0));
       tensor = layer->getOutput(0);
     }
-    tensor->setName(output.second.c_str());
-    VLOG(1) << "Marking output tensor " << output.first << ", as output tensor "
-            << output.second;
+    tensor->setName(output.dest_node_name.c_str());
+    // Set type after marking as output. TRT only supports setType for engine
```

Comment:
Do we need to set the type ***after*** marking it as output? Can we add a test for this fix as well?

Reply:
Yes, it must be after marking it as an output.
I think the only difference is that if you set the type before, TRT will log a warning. But they claim to only support setType for inputs and outputs so we shouldn't do that.

Reply:
Cool, thanks. If you can help to add a simple test using maybe TopK, to reproduce the problem (i.e. output dtype is kFLOAT before but kINT32 after), then we're good to go.

Reply:
@aaroey Thanks, I have added a test.